### ISSUE TITLE
Training time is too long

### ISSUE THREAD

**Issue by xingpeima** [ashawkey/RAD-NeRF#23]
Hi,
I found that it takes 0.07 seconds to train a step forward, but 15 seconds to backward，thus the total time will take several days.
Is there something wrong with my training code！

**Comment by ashawkey** [comment]
This is strange, what's the environment (e.g., GPU, CUDA, OS) you use? How do you measure the time?

**Comment by ashawkey** [comment]
No... it should take less than 10 hours to finish all epochs on V100. Are you facing this slow speed problem when training other DL models (e.g., resnet)?

**Comment by Erickrus** [comment]
Is there a way to speed up the training process, with only 1 GPU ?

**Comment by ashawkey** [comment]
The current training speed doesn't have much space to improve I guess. Maybe you could increase the `num_rays` and train less steps, but this may scarifice performance. Also, you may try torch 2.0's `compile`.

**Comment by yediny** [comment]
I have CUDA version 11.7, A100 40GB, and I'm having an issue where it's taking 4-5 hours per epoch.
![스크린샷 2023-02-21 오후 1 15 05](https://user-images.githubusercontent.com/69076426/220246349-729237b7-e61b-4a5a-a99f-616c1dcd77d3.png)
![스크린샷 2023-02-21 오후 1 13 59](https://user-images.githubusercontent.com/69076426/220246359-375f87ad-72a7-4c73-b20f-2f8c9993f826.png)

**Comment by ashawkey** [comment]
@yediny Could you provide the command you use? If you have enough GPU memory, you could try to use `--preload 2` to see if the speed bottleneck is image loading.

**Comment by yediny** [comment]
Even with preload applied.. it is still the same speed. 
Is there any way to make the most use of gpu memory for training? 
And under the normal case, does it take a day to training one model?
![스크린샷 2023-02-23 오후 2 00 45](https://user-images.githubusercontent.com/69076426/220825285-1c21deb2-c216-4d58-95ff-b6fe584f9253.png)

**Comment by braintown** [comment]
how to use the torch.compile accelerate training?

**Comment by braintown** [comment]
it seems we should use LightningModule to rewrite the nn.Module and then use the compile?
